pkg/api/api_test.go

```diff
@@ -45,3 +53,79 @@ func newTestServer(t *testing.T, o testServerOptions) *http.Client {
 		}),
 	}
 }
+
+func TestParseName(t *testing.T) {
```

Comment:
it would be nice to have a test where you explicitly try to parse a hash and assert that the resolver _was not_ called. a problem we had in old swarm is that we used to try to query and resolve normal hashes too. this was later discovered by infura, not us.

Reply:
Good point, current fallback (default) validator may still allow this behavior. The test for this, however, should be in the `resolver` package.